realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -168,10 +215,10 @@ public void onChange(Realm object) {
         });
     }
 
-    // Test that an async transaction that throws an exception propagate it properly to the user.
+    // Test that an async transaction that throws when call cancelTransaction manually.
     @Test
     @RunTestInLooperThread
-    public void executeTransactionAsync_exceptionHandling() throws Throwable {
+    public void executeTransactionAsync_cancelTransactionInside() throws Throwable {
```

Comment:
Looks like we don't rethrow the original error in the old implementation either. This test was mostly to check that we handled cancel+throw inside the closure. I re-added the exception since I think that was kinda the point